primitives/src/script/borrowed.rs

```diff
@@ -115,7 +115,7 @@ impl Script {
     pub fn is_empty(&self) -> bool { self.0.is_empty() }
 
     /// Converts a [`Box<Script>`](Box) into a [`ScriptBuf`] without copying or allocating.
-    #[must_use = "`self` will be dropped if the result is not used"]
+    #[must_use]
```

Comment:
Why? The previous message was fine.

Reply:
Sorry, I had complained about this message in another case (I think a `Copy` type, or maybe something that took a reference, or maybe something that consumed `self` but always dropped it and returned something else) and I think maybe Tobin was just blindly killing instances of this message. Since it's true in a fairly narrow set of circumstances.

FWIW I don't think it's a super useful message even when it's true. `self` will get consumed regardless of whether the return value is used, and if the user's code compiles then presumably they didn't actually need it.

Though I guess, if the code is *not* compiling, it's probably helpful for us to provide a nice warning here to hint to them what's wrong..

Reply:
See my other comment int the other PR.

Reply:
> `self` will get consumed regardless of whether the return value is used

IIRC @apoelstra pointed this out during dev of `must_use` stuff. And FTR I cargo-cult-copied this from `std` originally if my memory serves me correctly.